saleor/graphql/checkout/mutations.py

```diff
@@ -855,17 +856,33 @@ def perform_mutation(cls, _root, info, checkout_id, promo_code):
 class CheckoutUpdateMeta(UpdateMetaBaseMutation):
     class Meta:
         description = "Updates metadata for checkout."
-        permissions = (OrderPermissions.MANAGE_ORDERS,)
+        permissions = (OrderPermissions.MANAGE_ORDERS, CheckoutPermissions.MANAGE_CHECKOUTS, )
         model = models.Checkout
         public = True
         error_type_class = CheckoutError
         error_type_field = "checkout_errors"
 
+    class Arguments:
+        token = graphene.UUID(
+            description="Token of an object to update.",
+            required=True,
+        )
+
+        input = MetaInput(
+            description="Fields required to update new or stored metadata item.",
+            required=True,
+        )
+
+    @classmethod
+    def get_instance(cls, info, **data):
+        token = data["token"]
+        return models.Checkout.objects.get(token=token)
```

Comment:
This can raise `DoesNotExist` error. You could use `get_node_or_error` from BaseMutation to automatically handle it

Reply:
Gotcha! What arguments would I pass in? What does `node_id` refer to, the `token`?  Is `field_type` required also?

Reply:
Oh, my bad. I forgot that `checkout_token` is not translated in the same way as `ID`. In that case, I would recommend wrapping `.get` or  `filter().first` in `try except` block and raise `ValidationError("Couldn't resolve to a node: %s" % token, code="not_found")`